src/sentry/uptime/endpoints/organiation_uptime_alert_index.py

```diff
@@ -0,0 +1,119 @@
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import NoProjects
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.api.helpers.teams import get_teams
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.apidocs.constants import RESPONSE_FORBIDDEN, RESPONSE_NOT_FOUND, RESPONSE_UNAUTHORIZED
+from sentry.apidocs.parameters import GlobalParams, OrganizationParams, UptimeParams
+from sentry.apidocs.utils import inline_sentry_response_serializer
+from sentry.db.models.query import in_iexact
+from sentry.models.organization import Organization
+from sentry.search.utils import tokenize_query
+from sentry.types.actor import Actor
+from sentry.uptime.endpoints.serializers import (
+    ProjectUptimeSubscriptionSerializer,
+    ProjectUptimeSubscriptionSerializerResponse,
+)
+from sentry.uptime.models import ProjectUptimeSubscription
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Crons"])
+class OrganizationUptimeAlertIndexEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.CRONS
+    permission_classes = (OrganizationPermission,)
+
+    @extend_schema(
+        operation_id="Retrieve Uptime Alets for an Organization",
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+            OrganizationParams.PROJECT,
+            GlobalParams.ENVIRONMENT,
+            UptimeParams.OWNER,
+        ],
+        responses={
+            200: inline_sentry_response_serializer(
+                "UptimeAlertList", list[ProjectUptimeSubscriptionSerializerResponse]
+            ),
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def get(self, request: Request, organization: Organization) -> Response:
+        """
+        Lists uptime alerts. May be filtered to a project or environment.
+        """
+        try:
+            filter_params = self.get_filter_params(request, organization, date_filter_optional=True)
+        except NoProjects:
+            return self.respond([])
+
+        queryset = ProjectUptimeSubscription.objects.filter(
+            project__organization_id=organization.id, project_id__in=filter_params["project_id"]
+        )
+        query = request.GET.get("query")
+        owners = request.GET.getlist("owner")
+
+        if "environment" in filter_params:
+            queryset = queryset.filter(environment__in=filter_params["environment_objects"])
+
+        if owners:
+            owners = set(owners)
+
+            # Remove special values from owners, this can't be parsed as an Actor
+            include_myteams = "myteams" in owners
+            owners.discard("myteams")
+            include_unassigned = "unassigned" in owners
+            owners.discard("unassigned")
```

Comment:
I thought we could handle these? Since we know the current users teams and which teams own the uptime monitors?

Same for unassigned, wouldn't that be rows with no owners?

Reply:
it does handle these. This is actually almost verbatim copied from the crons listing endpint. We probably should generalize or provide some common utility for it 😬

Reply:
I just misread the comment, nm...